subsys/net/lib/sockets/sockets_net_mgmt.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdbool.h>
+#include <fcntl.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_sock_mgmt, CONFIG_NET_SOCKETS_LOG_LEVEL);
+
+#include <kernel.h>
+#include <misc/util.h>
+#include <net/socket.h>
+#include <syscall_handler.h>
+#include <misc/fdtable.h>
+#include <net/socket_net_mgmt.h>
+
+#include "sockets_internal.h"
+#include "net_private.h"
+
+#define MSG_ALLOC_TIMEOUT K_MSEC(100)
+
+struct net_mgmt_socket {
+	/* Network interface related to this socket */
+	struct net_if *iface;
+
+	/* A way to separate different sockets that listen same events */
+	u32_t pid;
+
+	/* net_mgmt mask */
+	u32_t mask;
+
+	/* Message allocation timeout */
+	s32_t alloc_timeout;
+
+	/* net_mgmt event timeout */
+	s32_t wait_timeout;
```

Comment:
from lines 79 and 80, these 2 attributes above don't seem necessary. At least not yet, they could be added if needed in the future.

Reply:
I was planning to add a way to have a proper timeout when waiting events, thus the timeout value here.